File: src/org/digitalcampus/oppia/utils/storage/StorageLocationInfo.java
package org.digitalcampus.oppia.utils.storage;

import android.content.Context;

import org.digitalcampus.mobile.learning.R;

public class StorageLocationInfo {

    public final String path;
    public final boolean readonly;
    public final boolean removable;
    public final int number;

    StorageLocationInfo(String path, boolean readonly, boolean removable, int number) {
        this.path = path;
        this.readonly = readonly;
        this.removable = removable;
        this.number = number;
    }

    public String getDisplayName(Context ctx) {
        StringBuilder res = new StringBuilder();
        if (!removable) {
            res.append(ctx.getString(R.string.storageNonRemovable));
        } else {
            res.append(ctx.getString(R.string.storageRemovable));
            if (number > 1) {
                res.append(" ").append(number);
            }
        }
        if (readonly) {
            res.append(" (Read only)");
        }
        return res.toString();
    }
}

File: app/src/main/java/org/digitalcampus/oppia/utils/storage/StorageLocationInfo.kt
package org.digitalcampus.oppia.utils.storage

import android.content.Context
import org.digitalcampus.mobile.learning.R
import org.digitalcampus.oppia.activity.PrefsActivity
import org.digitalcampus.oppia.utils.TextUtilsJava

class StorageLocationInfo internal constructor(val type: String, val path: String, val readonly: Boolean, val removable: Boolean, val number: Int) {
    fun getDisplayName(ctx: Context): String {
        return ctx.getString(if (TextUtilsJava.equals(type, PrefsActivity.STORAGE_OPTION_INTERNAL)) R.string.prefStorageOptionInternal else R.string.prefStorageOptionExternal)
    }
}